crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -1,86 +1,119 @@
 //! Module for establishing Websocket-based Interfaces,
 //! i.e. those configured with `InterfaceDriver::Websocket`
 
-use super::error::InterfaceError;
 use super::error::InterfaceResult;
 use crate::conductor::interface::*;
-use crate::conductor::manager::ManagedTaskResult;
 use crate::conductor::manager::TaskManagerClient;
-use futures::FutureExt;
 use holochain_serialized_bytes::SerializedBytes;
 use holochain_types::signal::Signal;
-use holochain_websocket::ListenerHandle;
-use holochain_websocket::ListenerItem;
+use holochain_websocket::ReceiveMessage;
 use holochain_websocket::WebsocketConfig;
 use holochain_websocket::WebsocketListener;
-use holochain_websocket::WebsocketMessage;
 use holochain_websocket::WebsocketReceiver;
 use holochain_websocket::WebsocketSender;
-use std::convert::TryFrom;
 
 use std::sync::Arc;
 use tokio::sync::broadcast;
 use tokio::task::JoinHandle;
-use tokio_stream::StreamExt;
 use tracing::*;
-use url2::url2;
+
+/// Concurrency count for websocket message processing.
+/// This could represent a significant memory investment for
+/// e.g. app installations, but we also need enough buffer
+/// to accomodate interdependent operations.
+const CONCURRENCY_COUNT: usize = 128;
 
 // TODO: This is arbitrary, choose reasonable size.
+// ERROR TODO XXX (david.b): There is no such thing as backpressure in
+//                           broadcast queues! It'll just start deleting
+//                           items, and giving "Lagged" errors on receivers.
 /// Number of signals in buffer before applying
 /// back pressure.
 pub(crate) const SIGNAL_BUFFER_SIZE: usize = 50;
 /// The maximum number of connections allowed to the admin interface
 pub const MAX_CONNECTIONS: usize = 400;
 
+/// For some reason, the "task manager" doesn't actually shut down tasks...
+/// We have to have this awkward workaround.
```

Comment:
```suggestion
// For some reason, the "task manager" doesn't actually shut down tasks...
// We have to have this awkward workaround.
```

Reply:
```suggestion
/// Adds the ability to write a task_motel task which simply gets aborted
/// when its stop signal is sent, rather than needing to handle the stop 
/// signal explicitly. Note that this requires an additional tokio spawned task.
```

Reply:
Changed the comment as part of the `tokio::select!` change, let me know if it works.